java.lang.UnsupportedOperationException when running without --plugin-path argument

### Plugin Modernizer version

1158.va_b_46a_2a_dde30

### What Operating System are you using

Linux

### Reproduction steps

plugin-modernizer run --recipe AddCodeOwner

```
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.jenkins.tools.pluginmodernizer.cli.Main.main(Main.java:47)
Caused by: java.lang.UnsupportedOperationException
        at java.base/java.util.ImmutableCollections.uoe(ImmutableCollections.java:142)
        at java.base/java.util.ImmutableCollections$AbstractImmutableCollection.add(ImmutableCollections.java:147)
        at io.jenkins.tools.pluginmodernizer.cli.options.PluginOptions.getEffectivePlugins(PluginOptions.java:81)
```

### Expected Results

No issue. SImilar to `plugin-modernizer run --plugin-path . --recipe AddCodeOwner`

Need probably need an integration test

--- a/plugin-modernizer-cli/src/main/java/io/jenkins/tools/pluginmodernizer/cli/options/PluginOptions.java
+++ b/plugin-modernizer-cli/src/main/java/io/jenkins/tools/pluginmodernizer/cli/options/PluginOptions.java
@@ -78,7 +78,7 @@ public final class PluginOptions implements IOption {
         // Use current folder as plugin if no plugin is provided
         if (effectivePlugins.isEmpty()) {
             try {
-                effectivePlugins.add(new PluginPathConverter().convert("."));
+                return List.of(new PluginPathConverter().convert("."));
             } catch (Exception e) {
                 throw new ModernizerException("Current directory doesn't seem to contains a plugin", e);
             }

--- a/plugin-modernizer-cli/src/test/java/io/jenkins/tools/pluginmodernizer/cli/options/PluginOptionsTest.java
+++ b/plugin-modernizer-cli/src/test/java/io/jenkins/tools/pluginmodernizer/cli/options/PluginOptionsTest.java
@@ -1,0 +1,53 @@
+package io.jenkins.tools.pluginmodernizer.cli.options;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import io.jenkins.tools.pluginmodernizer.core.config.Config;
+import io.jenkins.tools.pluginmodernizer.core.model.ModernizerException;
+import io.jenkins.tools.pluginmodernizer.core.model.Plugin;
+import java.lang.reflect.Field;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+/**
+ * Test for {@link PluginOptions}
+ */
+public class PluginOptionsTest {
+
+    @Test
+    public void testWithoutPluginSetToDefault() {
+        Config.Builder builder = Config.builder();
+        PluginOptions pluginOptions = new PluginOptions();
+        // Test
+        assertThrows(
+                ModernizerException.class,
+                () -> {
+                    pluginOptions.config(builder);
+                },
+                "Path does not contain a Jenkins plugin: .");
+    }
+
+    @Test
+    public void testWithOnePlugin() throws Exception {
+
+        Config.Builder builder = Config.builder();
+        PluginOptions pluginOptions = new PluginOptions();
+
+        // Set one plugin
+        Field field = ReflectionUtils.findFields(
+                        PluginOptions.class,
+                        f -> f.getName().equals("plugins"),
+                        ReflectionUtils.HierarchyTraversalMode.TOP_DOWN)
+                .get(0);
+        field.setAccessible(true);
+        field.set(pluginOptions, List.of(Plugin.build("one-plugin")));
+        pluginOptions.config(builder);
+
+        // Check only one plugin on the list
+        Config config = builder.build();
+        assertEquals(1, config.getPlugins().size(), "Only one plugin should be on the list");
+        assertEquals("one-plugin", config.getPlugins().get(0).getName(), "Plugin name should be 'one-plugin'");
+    }
+}

--- a/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/github/GHServiceTest.java
+++ b/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/github/GHServiceTest.java
@@ -1,9 +1,21 @@
 package io.jenkins.tools.pluginmodernizer.core.github;
 
-import static org.junit.jupiter.api.Assertions.*;
-import static org.mockito.ArgumentMatchers.*;
-import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
 import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 
 import com.google.inject.Guice;
 import io.jenkins.tools.pluginmodernizer.core.GuiceModule;

